app/jobs/zipmaker_job.rb

```diff
@@ -4,13 +4,25 @@
 # If needed, zip files to zip storage and calculate checksum(s).
 # Otherwise, touch the existing main ".zip" file to freshen it in cache.
 # Invoke PlexerJob for each zip part.
-class ZipmakerJob < DruidVersionJobBase
+class ZipmakerJob < ApplicationJob
   queue_as :zipmaker
   delegate :create_zip!, :file_path, :part_keys, to: :zip
 
+  attr_accessor :zip
+
+  before_perform do |job|
+    job.zip = DruidVersionZip.new(job.arguments.first, job.arguments.second, job.arguments.third)
+  end
+
+  # Does queue locking on ONLY druid and version (as first and second parameters)
+  def self.lock(*args)
+    "lock:#{name}-#{args.slice(0..1)}"
+  end
+
   # @param [String] druid
   # @param [Integer] version
-  def perform(druid, version)
+  # @param [String] moab_replication_path The path containing the druid tree from which the zipped version should be created
+  def perform(druid, version, moab_replication_path) # rubocop:disable Lint/UnusedMethodArgument
```

Comment:
Please clarify in the job that the `moab_replication_path` argument *is* in fact used (on line 14 as `job.arguments.third`).